aircmd/models/base.py

```diff
@@ -45,50 +44,143 @@ class GlobalSettings(BaseSettings):
+        + glob("**/.pytest_cache", recursive=True)
+        + glob("**/.eggs", recursive=True)
+        + glob("**/.mypy_cache", recursive=True)
+        + glob("**/.DS_Store", recursive=True)
+    , env="DEFAULT_EXCLUDED_FILES") 
+    DOCKER_VERSION:str = Field("20.10.23", env="DOCKER_VERSION")
+    DOCKER_DIND_IMAGE: str = Field("docker:20-dind", env="DOCKER_DIND_IMAGE")
+    DOCKER_CLI_IMAGE: str = Field("docker:20-cli", env="DOCKER_CLI_IMAGE")
+    GRADLE_CACHE_PATH: str = Field("/root/.gradle/caches", env="GRADLE_CACHE_PATH")
+    GRADLE_BUILD_CACHE_PATH: str = Field(f"{GRADLE_CACHE_PATH}/build-cache-1", env="GRADLE_BUILD_CACHE_PATH")
+    GRADLE_READ_ONLY_DEPENDENCY_CACHE_PATH: str = Field("/root/gradle_dependency_cache", env="GRADLE_READ_ONLY_DEPENDENCY_CACHE_PATH")
+
+    SECRET_DOCKER_HUB_USERNAME: Optional[str] = Field(None, env="SECRET_DOCKER_HUB_USERNAME")
+    SECRET_DOCKER_HUB_PASSWORD: Optional[str] = Field(None, env="SECRET_DOCKER_HUB_PASSWORD")
+    
 
     class Config:                                                                                                                                                        
          arbitrary_types_allowed = True                                                                                                                                   
          env_file = '.env' 
 
+
+'''
+In the case of modeling CI pipelines, the child-knows-parent approach seems more suitable for the 
+following reasons:                                                                                
+
+ 1 Analyzing individual pipeline runs: When you want to analyze a specific pipeline run, you might
+   need to compare it with its parent or ancestor runs to understand the differences or           
+   improvements. In this case, having a reference to the parent pipeline makes it easier to       
+   traverse up the pipeline hierarchy.                                                            
+ 2 Dependency tracking: In some CI systems, a pipeline run might depend on the successful         
+   completion of its parent or ancestor runs. With the child-knows-parent approach, you can easily
+   check the status of the parent pipelines and determine if the current pipeline run should      
+   proceed or not.                                                                                
+ 3 Debugging and troubleshooting: When a pipeline run fails, you might need to investigate the    
+   cause of the failure by looking at the parent or ancestor runs. The child-knows-parent approach
+   allows you to easily navigate up the pipeline hierarchy to find the relevant information.      
+
+While it's true that some use cases might require traversing the pipeline hierarchy from the root 
+to the leaves, these cases are less common in CI systems. The child-knows-parent approach provides
+more flexibility and ease of use for the typical operations performed on CI pipelines.        
```

Comment:
Just a little friendly push back on 
> child-knows-parent approach seems more suitable
---
> 1. Analyzing individual pipeline runs
Can you say more about what we are analyzing?

For example in the airbyte-ci pipeline we take a parent-knows-child approach and upload logs in a heirarchical way where if a 10 connectors are run in a nightly ci test then youll see a report for the nightly command at `gs://airbyte-ci-reports/nightly/123/output.json` and then another report for each connector `gs://airbyte-ci-reports/nightly/123/source-s3/1.2.3/output.json`

This lets us analyze the results of the nightly pipeline easier

(also these arent the actual paths but you get the idea)

> 2 Dependency tracking: a pipeline run might depend on the successful         
   completion of its parent or ancestor runs

 I think air-traffic control has to be the responsibility of the parent.

 Sub tasks are often unaware of their siblings by design and its up to the parent to determine when each task is run, in what order, which are paralell and which can be skipped. I cant see a child successfully making those determinations with out the parent getting very involved. 
 


-------

Though @cpdeethree I may very well be missing or mis-intrepreting something here!

Reply:
I am not 100% sold on this, I was thinking about things from an analytics perspective (like how we would be working with the data) as I added that justification. There’s also no reason why I couldn’t add references in the other direction when exporting for analytics though. You make some pretty compelling arguments! Curious what Augustin and helder think as well!

Reply:
Thanks for being open! I think this should be a good seed for a fun discussion today.

Also worthe restating: I love what you've got here 😍

Reply:
Yes, I think having `steps` in the Pipeline is easier to understand. Each step would be sequential, steps with a list of pipelines could be concurrent.

Reply:
After some discussion with augustin we agree. I will update this model today. Thanks for the great feedback from everyone

Reply:
After some further work on this, I've decided to move this work to a separate pr, which I will link to here when it's open, in order to keep the scope of this one relatively sane